src/m365/spo/commands/navigation/navigation-node-add.ts

```diff
@@ -147,13 +154,58 @@ class SpoNavigationNodeAddCommand extends SpoCommand {
     };
 
     try {
-      const res = await request.post<any>(requestOptions);
+      const res = await request.post<NavigationNode>(requestOptions);
+
+      if (args.options.openInNewWindow) {
+        if (this.verbose) {
+          logger.logToStderr(`Making sure that the newly added navigation node opens in a new window.`);
+        }
+
+        let menuState: MenuState | undefined;
+        let menuStateItem: MenuStateNode | undefined;
+        if (args.options.parentNodeId) {
+          menuState = await spo.getQuickLaunchMenuState(args.options.webUrl);
+          menuStateItem = this.getMenuStateNode(menuState.Nodes, res.Id.toString());
+          if (!menuStateItem) {
+            menuState = await spo.getTopNavigationMenuState(args.options.webUrl);
+            menuStateItem = this.getMenuStateNode(menuState.Nodes, res.Id.toString());
+          }
+        }
+        else {
+          if (args.options.location === 'QuickLaunch') {
+            menuState = await spo.getQuickLaunchMenuState(args.options.webUrl);
+            menuStateItem = this.getMenuStateNode(menuState.Nodes, res.Id.toString());
+          }
+          else {
+            menuState = await spo.getTopNavigationMenuState(args.options.webUrl);
+            menuStateItem = this.getMenuStateNode(menuState.Nodes, res.Id.toString());
+          }
+        }
+        menuStateItem!.OpenInNewWindow = true;
+        await spo.saveMenuState(args.options.webUrl, menuState!);
```

Comment:
What if we refactor the approach a bit to have less code to manage 🤔 
@MathijsVerbeeck I tried the below implementation. I think it will do the same thing but maybe I am missing some edge case. Could you double-check?

```suggestion
        const id: string = res.Id.toString();
        let menuState: MenuState | undefined = args.options.location === 'TopNavigationBar' ?
          await spo.getTopNavigationMenuState(args.options.webUrl) :
          await spo.getQuickLaunchMenuState(args.options.webUrl);
        let menuStateItem: MenuStateNode | undefined = this.getMenuStateNode(menuState.Nodes, id);

        if (args.options.parentNodeId && !menuStateItem) {
          menuState = await spo.getTopNavigationMenuState(args.options.webUrl);
          menuStateItem = this.getMenuStateNode(menuState.Nodes, id);
        }

        menuStateItem!.OpenInNewWindow = true;
        await spo.saveMenuState(args.options.webUrl, menuState!);
```

Reply:
This makes it much more readable indeed, great suggestion.

Reply:
> if (args.options.parentNodeId && !menuStateItem) {
>           menuState = await spo.getTopNavigationMenuState(args.options.webUrl);
>           menuStateItem = this.getMenuStateNode(menuState.Nodes, id);
>         }

If I'm not mistaken and read your suggestion properly, this code can also be removed.